Doc/whatsnew/3.14.rst

```diff
@@ -357,6 +357,10 @@ New Features
 
   (Contributed by Victor Stinner in :gh:`119182`.)
 
+* Add :c:func:`PyBytes_Join(seq, iterable) <PyBytes_Join>` function,
```

Comment:
How is it rendered?

Reply:
It will be a clickable `PyBytes_Join(seq, iterable)` pointing to the function `PyBytes_Join`:

![image](https://github.com/user-attachments/assets/160a92eb-1a2f-4f50-8a45-67a4e60c80af)

More generally, you can have:

```rst
:role:`whatever I want as a title <reference>`
```

It's the same idea as for `[link title](url)` in markdown.

Reply:
I just worried that it may be rendered as `PyBytes_Join(seq, iterable)()`. The `c:func` role usually adds `()`.

Reply:
Yes, by default the () are added (but I think you can suppress it using some configuration value). Whenever you want to render something yourself, you can use this alternative syntax!

Reply:
See https://cpython-previews--121646.org.readthedocs.build/en/121646/whatsnew/3.14.html#id4 for the rendered doc.

See "📚 Documentation preview 📚: https://cpython-previews--121646.org.readthedocs.build/ " in the PR description.